src/theme/interaction.rs

```diff
@@ -1,16 +1,21 @@
-use bevy::{ecs::system::SystemId, prelude::*};
-
-use crate::{assets::SfxHandles, audio::sfx::SfxCommands as _};
+use crate::{asset_tracking::LoadResource, audio::SoundEffect, screens::Screen};
+use bevy::{
+    ecs::{system::SystemId, world::Command},
+    prelude::*,
+};
+use std::{collections::VecDeque, marker::PhantomData};
 
 pub(super) fn plugin(app: &mut App) {
     app.register_type::<InteractionPalette>();
+    app.load_resource::<ButtonAssets>();
     app.add_systems(
         Update,
         (
             trigger_on_press,
             apply_interaction_palette,
             trigger_interaction_sfx,
-        ),
+        )
+            .run_if(not(in_state(Screen::Loading)).and_then(not(in_state(Screen::Splash)))),
```

Comment:
I'm curious why this run condition is needed. Ideally, `src/theme/` shouldn't depend on `src/screens/`.

Reply:
This exists to ensure it doesn't run before the `ButtonAssets` resource (and associated assets) have been loaded. The asset loading lifecycle is currently tied to `Screen`. 

An alternative would be to turn the `Res<ButtonAssets>` in `trigger_interaction_sfx` into `Option<Res<ButtonAssets>>` to allow the system to run panic-free when the resource doesn't exist yet. However conceptually, I think tying this into the current (or a separate) asset loading lifecycle is the right move.

Reply:
I've switched to `resource_exists` here